sdk/servicebus/service-bus/src/serviceBusError.ts

```diff
@@ -0,0 +1,179 @@
+  /**
+   * The lock on the message is lost. Callers should call attempt to receive and process the message again.
+   */
+  | "MessageLockLost"
+  /**
+   * The requested message was not found.
+   */
+  | "MessageNotFound"
+  /**
+   * A message is larger than the maximum size allowed for its transport.
+   */
+  | "MessageSizeExceeded"
+  /**
+   * An entity with the same name exists under the same namespace.
+   */
+  | "MessagingEntityAlreadyExists"
+  /**
+   * The Messaging Entity is disabled. Enable the entity again using Portal.
+   */
+  | "MessagingEntityDisabled"
+  /**
+   * The quota applied to an Service Bus resource has been exceeded while interacting with the Azure Service Bus service.
+   */
+  | "QuotaExceeded"
+  /**
+   * The Azure Service Bus service reports that it is busy in response to a client request to perform an operation.
+   */
+  | "ServiceBusy"
+  /**
+   * An operation or other request timed out while interacting with the Azure Service Bus service.
+   */
+  | "ServiceTimeout"
+  /**
+   * There was a general communications error encountered when interacting with the Azure Service Bus service.
+   */
+  | "ServiceCommunicationProblem"
+  /**
+   * The requested session cannot be locked.
+   */
+  | "SessionCannotBeLocked"
+  /**
+   * The lock on the session has expired. Callers should request the session again.
+   */
+  | "SessionLockLost"
+  /**
+   * The user doesn't have access to the entity.
+   */
+  | "Unauthorized";
+
+/**
+ * Translation between the MessagingErrorCodes into a ServiceBusReason
+ *
+ * @internal
+ * @ignore
+ */
+export const wellKnownMessageCodesToServiceBusReasons: Map<
+  MessagingErrorCodes,
+  ServiceBusErrorReason
+> = new Map([
+  ["MessagingEntityNotFoundError", "MessagingEntityNotFound"],
+  ["MessageLockLostError", "MessageLockLost"],
+  ["MessageNotFoundError", "MessageNotFound"],
+  ["MessageTooLargeError", "MessageSizeExceeded"],
+  ["MessagingEntityAlreadyExistsError", "MessagingEntityAlreadyExists"],
+  ["MessagingEntityDisabledError", "MessagingEntityDisabled"],
+  ["QuotaExceededError", "QuotaExceeded"],
+  ["ServerBusyError", "ServiceBusy"],
+
+  // not sure about these two.
+  ["MessageWaitTimeout", "ServiceTimeout"],
+  ["OperationTimeoutError", "ServiceTimeout"],
+  ["ServiceCommunicationError", "ServiceCommunicationProblem"],
+  ["SessionCannotBeLockedError", "SessionCannotBeLocked"],
+  ["SessionLockLostError", "SessionLockLost"],
+  ["UnauthorizedError", "Unauthorized"]
+]);
+
+/**
+ * Errors that occur within Service Bus
+ */
+export class ServiceBusError extends MessagingError {
+  /**
+   * The reason for the failure.
+   *
+   * **GeneralError**: The exception was the result of a general error within the client library.
+   * **MessagingEntityNotFound**: A Service Bus resource cannot be found by the Service Bus service.
+   * **MessageLockLost**: The lock on the message is lost. Callers should call attempt to receive and process the message again.
```

Comment:
```suggestion
   * **MessageLockLost**: The lock on the message is lost. Callers should attempt to receive and process the message again.
```

Reply:
CC: @JoshLove-msft . This text is directly from your error codes. :)